benchmarks/stress.py

```diff
@@ -0,0 +1,214 @@
+
+    if dest:
+        out += "; Output located on targeted hosts at: {}".format(dest)
+
+    result['changed']   = True
+    result['exec_cmd']  = benchmark_command
+    result['stdout']    = out.rstrip("\r\n")
+    result['stderr']    = err.rstrip("\r\n")
+    result['rc']        = rc
+
+def parse(module, result, params):
+
+    dest        = params['dest']
+
+    if not os.path.exists(dest):
+        module.fail_json(msg="{} does not exist".format(dest))
+
+    result_file = open(dest, 'r')
+    data = result_file.readlines()
+    result_file.close()
+
+    json_result = dict()
+
+    for line in data:
+        if "hogs" in line:
+            match = re.search('(\d+)\s+cpu,\s+(\d+)\s+io,\s+(\d+)\s+vm,\s+(\d+)\s+hdd', line)
+            if match:
+                json_result['cpu_hogs'] = match.group(1)
+                json_result['io_hogs'] = match.group(2)
+                json_result['vm_hogs'] = match.group(3)
+                json_result['hdd_hogs'] = match.group(4)
+        elif "successful" in line:
+            match = re.search('completed\s+in\s+(\d+)', line)
+            if match:
+                json_result['completion_time'] = match.group(1)
+    if not json_result:
+        module.fail_json(msg="Invalid result file at {}: {}".format(dest, line))
+
+    result['changed'] = True
+    result['results'] = json.dumps(json_result)
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            chdir=dict(required=False,
+                       type='str'),
+            cpu=dict(required=False,
+                     type='int'),
+            delay=dict(required=False,
+                       type='int'),
+            dest=dict(required=False,
+                      type='str',
+                      default=None),
+            executable=dict(required=False,
+                            type='str'),
+            state=dict(type='str',
+                       default="benchmark",
+                       choices=['benchmark', 'parse']),
+            timeout=dict(required=False,
+                         type='int'),
+
+        ),
+        supports_check_mode=False
+    )
+
+    stress_bin = None
+    result = {'changed': False, 'bench_config': dict()}
+
+    if module.params['state'] == 'parse':
+        if not module.params['dest']:
+            module.fail_json(msg='dest= is required for state=parse')
+        parse(module, result, module.params)
+
+    if module.params['state'] == 'benchmark':
+        missing_params = list()
+        for param in ['timeout', 'cpu']:
+            if not module.params[param]:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            module.fail_json(msg="missing required arguments: {}".format(missing_params))
+        if module.params['executable']:
+            stress_bin = module.params['executable']
+        else:
+            stress_bin = module.get_bin_path('stress', True, ['/usr/bin/local'])
```

Comment:
I believe that parameter is opt_dirs which just adds '/usr/bin/local' to the list of pre-defined paths to search. So I don't think there should be too much of a problem here.

[https://github.com/ansible/ansible/blob/fda933723c8cb60248d41b5691edd86fe0a40807/lib/ansible/module_utils/basic.py#L1723-L1728]